### DISCUSSION TITLE
Can we use a custom user's field in a filter?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#15152] [Archive]
Wth a custom field `company ` added to the `directus_users` collection, I'm trying to set a preset to show only the members of the company of the current user.
The filter `company `_eq `$CURRENT_USER.company` gives an empty result.

I thought this should work as in  [#10927](https://github.com/directus/directus/discussions/10927#discussioncomment-1936795)

**Comment by azrikahar** [comment]
> I thought this should work as in [#10927](https://github.com/directus/directus/discussions/10927#discussioncomment-1936795)

That's Permissions [Filter Rule](https://docs.directus.io/reference/filter-rules.html) (emphasis on Permissions), not filter. But I can see how it can be confused as such.

Nested dynamic variables currently does not work for filter, as described here in the notes: https://docs.directus.io/reference/filter-rules.html#dynamic-variables

> Note: This feature is only available for permissions, validation, and presets. Regular filters and conditional fields currently only support the root ID.

**Reply by Pachat** [reply]
>This feature is only available for permissions, validation, <div style="font-size: 24px">and **presets**</div>

My understanding is that it should work on presets.
" company _eq $CURRENT_USER.company " was just my way to write the preset filter `company  Equals $CURRENT_USER.company`
That filter is in preset not a regular filter.

On the other hand, I set a custom permission filter for the role on this collection 
```
{
	"_and": [
		{
			"company": {
				"_eq": "$CURRENT_USER.company"
			}
		}
	]
}
```
and that was not working either: the list stays empty.

If I hard code, replacing $CURRENT_USER.company by the real current user's company, the filter works for that user (and any user of that company).

In short, I did not succeed to use $CURRENT_USER.company, neither in the Permission read, nor in a Preset filter.


_I am now on 9.16.0_

**Reply by azrikahar** [reply]
> My understanding is that it should work on presets.

The permissions, validation and presets in that note should mean these Field Permissions, Field Validation and Field Presets 👍 Not to be confused with Presets & Bookmarks (once again I'm assuming you meant Presets & Bookmarks since you mentioned you wanted to only show certain items, whereas Field Presets is used to set default values.)

![image](https://user-images.githubusercontent.com/42867097/185535301-54d9fd2b-02d6-4a68-93b4-3fe6a728880d.png)

That being said, the custom permission filter should work, so something else might be up in your scenario 🤔 Here's a quick example I did:

https://user-images.githubusercontent.com/42867097/185539113-1931a2ea-7bef-49f2-8013-84733a8c5a06.mp4

We can see the `regular@example.com` user in `Regular` role, which is linked to `Company A`, can only see records related to that company. Perhaps you can refer and see what could be the difference between our setups?

**Reply by Pachat** [reply]
Thank you so much for your time.
I did a fresh installation and it works as expected :)

**Reply by lucastoneatto** [reply]
How can I filter users (in Directus users view) who are from the same tenant? Now if I want to apply a custom filter to the collection, it tells me that by default it already applies a filter that only lets me see my user.
How can I solve that?
![image](https://github.com/directus/directus/assets/5579126/df670fa6-9f87-4aeb-a372-b83c30faaccd)

**Comment by Pachat** [comment]
AFAIK this is just an example of the minimal filter: the user has to be allowed to read its item.
To apply a custom filter, just use the regular filter builder that is above.
As far as the current user is included in your filter, it will be OK.

**Reply by lucastoneatto** [reply]
but the final result is not what was sought.
I want to see all the users that belong to my tenant, and that default filter always shows me my own user and not anyone else's. That filter that is by default does not make any sense, because when you put all access to views, you can see all the users even from other tenants.
So, when I want to apply a custom mode to directus_users it does not work for a tenant model, because either you see all the users in the system, or you only see your user.
I hope the case I present is understood.

**Reply by Pachat** [reply]
`Suppose` you added a Tenant field to Directus_users
You just have to set, per tenant role on the Directus_users collection a filter on the Tenant field value.
(forget $CURRENT_USER or add it but under a OR statement)
```
OR
    ID = $CURRENT_USER
    tenant = 'Tenant 1'    (or, better ids if you use a relational table)
```

**Reply by lucastoneatto** [reply]
I can't. The filter is by default.

![image](https://github.com/directus/directus/assets/5579126/6c9e9050-3d05-437d-8352-86fa3807baa7)

**Reply by lucastoneatto** [reply]
I kept testing, and that hard filter that is applied by default (and that cannot be omitted) causes a lot of problems for a tenant application
If I create an Owner profile of a tenant, it can associate users with the tenant, but then it cannot remove them from the tenant because it only has permission to edit its own user, not the others'.
I think it would be good for this filter not to be hardcoded by the platform but rather to be a preloaded filter with its corresponding warning, just like it is now, but that you can remove it if it does not apply to your data model

**Reply by SadmanYasar** [reply]
Facing this exact issue. Did you find any solution to this?
